functorch/experimental/_map.py

```diff
@@ -80,17 +99,27 @@ def from_fun(t):
                     else:
                         # clone of a functional tensor produces a functional tensor
                         # but we want to avoid it so we clone a non-functional version
-                        maybe_unfunc_t = torch._functorch.aot_autograd.from_fun(t)
+                        maybe_unfunc_t = t
+                        if isinstance(t, FunctionalTensor):
+                            torch._sync(t)
+                            maybe_unfunc_t = from_fun(t)
+                        elif torch._is_functional_tensor(t):
+                            # need to handle both types of functionalization here:
+                            # these are the tensors that came from the user,
+                            # which could be either FunctionalTensorWrapper or FunctionalTensor
+                            torch._sync(t)
+                            maybe_unfunc_t = torch._from_functional_tensor(t)
```

Comment:
I'm confused about why we need to handle both FunctionalTensor here and C++ Functionalization. The way we actually hit this code is through AOTAutograd. I expect us to move AOTAutograd to Python Functionalization (not sure where this is in your stack). So we shouldn't need to have a branch for C++ Functionalization anymore

Reply:
Ah ok - I wasn't sure if this code was only meant to be used in AOTAutograd, or also might be used more generically, so I played it safe and just made it work with both flavors.

What do you think of leaving as-is in this PR, and me removing the C++ functionalization handling in the PR where I actually flip the switch on python functionalization in AOTAutograd?

I mainly tried to set the PR's up in that way so that there was as little code as possible in the "turn on python functionalization in AOTAutograd", since that PR carries slightly higher revert risk.

Reply:
Sounds fine